R/method-differentialAbundance.R

```diff
@@ -172,7 +201,25 @@ function(data, covariate, groupA, groupB, method = c("Maaslin2", "DESeq2"), verb
     comparator <- buildBinaryComparator(covariate, 'groupA', 'groupB')
 
     ## microbiomeComputations will remove for us rows not in either group, and provide validation
-    return(microbiomeComputations::differentialAbundance(data, comparator = comparator, method = method, verbose = verbose))
+    return(microbiomeComputations::internalDiffAbund(data, comparator = comparator, method = method, verbose = verbose))
+})
+
+#' @rdname differentialAbundance-methods
+#' @aliases differentialAbundance,CollectionWithMetadata,character,character,character-method
+setMethod("differentialAbundance", signature("CollectionWithMetadata", "character", "character", "character"),
+function(data, covariate, groupA, groupB, method = c("Maaslin2", "DESeq2"), verbose = c(TRUE, FALSE)) {
+    method <- veupathUtils::matchArg(method)
+    verbose <- veupathUtils::matchArg(verbose)
+
+    covariateDataType <- class(data@sampleMetadata@data[[covariate]])
+    if (!covariateDataType %in% c("factor", "character")) {
+        stop("Arguments 'groupA' and 'groupB' must be functions when 'covariate' is not a factor or character")
+    }
+
```

Comment:
We could consider adding a check here to make sure groupA and B are actually in the covariate var

Reply:
We should also check to make sure groupA and B don't overlap if that isn't done somehow in the depths of internalDiffAbund